player/screenshot.c

```diff
@@ -482,17 +483,19 @@ void cmd_screenshot(void *p)
         }
     }
 
-    cmd->success = false;
-
     struct image_writer_opts *opts = mpctx->opts->screenshot_image_opts;
     bool high_depth = image_writer_high_depth(opts);
 
     struct mp_image *image = screenshot_get(mpctx, mode, high_depth);
 
     if (image) {
         char *filename = gen_fname(cmd, image_writer_file_ext(opts));
-        if (filename)
-            cmd->success = write_screenshot(cmd, image, filename, NULL);
+        if (filename && write_screenshot(cmd, image, filename, NULL)) {
+            node_init(res, MPV_FORMAT_NODE_MAP, NULL);
+            node_map_add_string(res, "filename", filename);
+        } else {
+            cmd->success = false;
+        }
```

Comment:
You should be saving the result of `write_screenshot` to cmd->success no matter what.

Reply:
Is this a logic bug or just a stylistic thing ?
Because if anything fails, success is false, and it's true by default (player/command.h:52)

Reply:
Mostly just style. Better to be explicit imo than silently rely on this being true.

Reply:
Speaking of, we don't show the error message when either writing or generating the path fails.

Reply:
Are you sure? `gen_fname` should handle this already.

Reply:
oh, you're right, sorry for not noticing.